protocol/synthetix/contracts/modules/core/LiquidationModule.sol

```diff
@@ -92,9 +91,20 @@ contract LiquidationModule is ILiquidationModule {
             revert MustBeVaultLiquidated();
         }
 
+        // distribute any outstanding rewards distributor value to the user who is about to be liquidated, since technically they are eligible.
+        Pool.load(poolId).updateRewardsToVaults(
+            Vault.PositionSelector(accountId, poolId, collateralType)
+        );
+
         // This will clear the user's account the same way as if they had withdrawn normally
         epoch.updateAccountPosition(accountId, 0, 0);
 
+        // in case the liquidation caused the user to have less collateral than is actually locked in their account,
+        // this will ensure their locks are good.
+        // NOTE: limit is set to 50 here to prevent the user from DoSsing their account liquidation by creating locks on their own account
+        // if the limit is surpassed, their locks wont be scaled upon liquidation and that is their problem
+        Account.load(accountId).cleanAccountLocks(collateralType, 0, 50);
```

Comment:
we have to iterate on each lock because we don't have the multiplicative scalar necessary to update only one value.

we could possibly add a scalar/use a `ScalableMapping` to allow for all the locks to be reduced but 1) this would increase the gas of any operations that act upon locks because now they need to read and possibly write to the scalar, and 2) this is such an edgy edge case operation I decided it would be best to stick with an overall much simpler approach